analyses/cell-type-ewings/aucell-singler-annotation.sh

```diff
@@ -0,0 +1,168 @@
+        marker_gene_results="${sample_results_dir}/${library_id}_marker-gene-classifications.tsv"
+
+        # only run AUCell if sample is NOT the ref sample
+        if [[ ($sample_id != $ref_sample) && ($library_id != $ref_library) && (! -f $auc_results) ]]; then
+
+            echo "Running AUCell for sample: $sample_id and library: $library_id"
+
+            # run AUCell
+            Rscript $scripts_dir/01-run-aucell.R \
+                --sce_file $sce_file \
+                --auc_threshold $auc_threshold \
+                --output_file $auc_results \
+                --threads $threads
+
+        fi
+
+        if [[ ! -f $geneset_results ]]; then
+            # generate gene set scores
+            Rscript $scripts_dir/02-calculate-gene-set-scores.R \
+              --sce_file $sce_file \
+              --output_file $geneset_results
+        fi
+
+        if [[ ! -f "$sample_results_dir/${library_id}_aucell-report.html" ]]; then
+
+            # render notebook
+            Rscript -e "rmarkdown::render('$notebook_dir/01-auc-results.Rmd', \
+              clean = TRUE, \
+              output_dir = '$sample_results_dir', \
+              output_file = '${library_id}_aucell-report.html', \
+              params = list(library_id = '$library_id', \
+                            sce_file = '$sce_file', \
+                            aucell_results_file = '$auc_results', \
+                            ref_auc_results = '$ref_auc_results', \
+                            geneset_scores_file = '$geneset_results', \
+                            marker_genes_file = '$tumor_marker_genes_file', \
+                            marker_class_file = '${marker_gene_results}', \
+                            auc_threshold = $auc_threshold), \
+              envir = new.env()) \
+            "
+        fi
+    done
+done
+
+# SingleR -------------------------------------------------------------
+
+singler_ref_file="scratch/tumor-ref-singler.rds"
+
+if [[ ! -f $singler_ref_file ]]; then
+    echo "Creating tumor cell reference for SingleR"
+    # first build tumor reference using aucell outputs
+    Rscript $scripts_dir/03-generate-tumor-ref.R \
+      --data_dir "$data_dir" \
+      --auc_results_dir "$results_dir" \
+      --output_reference_file "$singler_ref_file"
+fi
+
+# now run SingleR and generate report for all samples
+for sample_id in $sample_ids; do
+
+    # define sample results directory
+    sample_results_dir="${results_dir}/${sample_id}"
+
+    for sce_file in $data_dir/$sample_id/*_processed.rds; do
+
+        # define library ID
+        library_id=$(basename $sce_file | sed 's/_processed.rds$//')
+
+        # define library specific input/output files
+        geneset_results="${sample_results_dir}/${library_id}_gene-set-scores.tsv"
+        singler_results="${sample_results_dir}/${library_id}_singler-classifications.tsv"
+
+        # only run SingleR if results don't already exist
+        if [[ ! -f $singler_results ]]; then
```

Comment:
I'd probably remove this `if`; I would be a bit worried about accidentally not updating these. I suppose you could add something to check if the ref file was modified more recently than the output, but at that point you are basically getting into writing a workflow manager. If you really think you need those features, I would probably go ahead and just turn this into a Nextflow workflow (with minimal config so it just runs locally)!

Reply:
This was really just to save time whenever I had to restart because of a failure. I can remove this before merging in.